File: immiglensBE/app/schemas/stats.py
from pydantic import BaseModel


class CaptureBreakdownItem(BaseModel):
    name: str
    value: int
    color: str


class EmployerBreakdownItem(BaseModel):
    name: str
    positions: int
    screenshots: int
    failed: int


class RoundTimelineItem(BaseModel):
    date: str
    completed: int
    pending: int
    failed: int


class DashboardStats(BaseModel):
    total_employers: int
    total_positions: int
    total_job_postings: int
    total_capture_rounds: int
    completed_rounds: int
    pending_rounds: int
    total_screenshots: int
    failed_screenshots: int
    capture_breakdown: list[CaptureBreakdownItem]
    employer_breakdown: list[EmployerBreakdownItem]
    rounds_timeline: list[RoundTimelineItem]

File: immiglensFE/src/types/stats.ts
export interface CaptureBreakdownItem {
  name: string
  value: number
  color: string
}

export interface EmployerBreakdownItem {
  name: string
  positions: number
  screenshots: number
  failed: number
}

export interface RoundTimelineItem {
  date: string
  completed: number
  pending: number
  failed: number
}

export interface DashboardStats {
  total_employers: number
  active_employers: number
  total_positions: number
  active_positions: number
  total_job_postings: number
  active_postings: number
  total_capture_rounds: number
  completed_rounds: number
  pending_rounds: number
  total_screenshots: number
  failed_screenshots: number
  capture_breakdown: CaptureBreakdownItem[]
  employer_breakdown: EmployerBreakdownItem[]
  rounds_timeline: RoundTimelineItem[]
}
